### PR TITLE
fix(inline-control-group): remove bottom spacing from last child control

### PR DESCRIPTION
**Pull request by SiTaggart**
Taking over from https://github.com/twilio-labs/paste/pull/941

Using stack to create the spacing and general tidy up. Also adding stories to verify no regressions.

### CODE REVIEW

**Review by TheSisb — APPROVED** [review]

**Comment by SiTaggart** [comment]
This is not as easy as first thought. The reason for the bottom margin is for when a group is orientated in a horizontal row and the number of items exceeds the row width space.

image

We can't remove the bottom margin in horizontal layouts entirely as the second row would remove the desired space between rows.

We can't _only_ remove the bottom margin on the last child, because as you can see in the screen shot, there might be more than one dangling options which will introduce the gap along the bottom of the groups.

I'll think about this some more. Maybe, because a group must have a legend all the spacing is top only?

**Comment by TheSisb** [comment]
I kind of like a rule of marginTop only on our components, with the option to unset the margin. Mixing the two leads to scenarios where the margins collapse sometimes. We also can't predict what comes after an element, but we may have an opinion on what should come before.

File: packages/paste-core/components/form/stories/form.stories.tsx [hunk 1]
```diff
@@ -65,8 +240,180 @@ export const All = (): React.ReactNode => {
       <Box marginBottom="space70">
         <FormLabel htmlFor={inputIDSeed('errorinput')}>Text Input</FormLabel>
         <FormInput hasError id={inputIDSeed('errorinput')} type="text" value="" />
+        <Grid>
+          <Column>
+            <CheckboxGroup
```
Comment (TheSisb) [inline packages/paste-core/components/form/stories/form.stories.tsx:245]: Seems like a lot of copy paste in this story. Can we component-ize this (like you did in inline-control-group) so the difference is easy to see?

File: .changeset/two-humans-press.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+'@twilio-paste/inline-control-group': patch
+'@twilio-paste/core': patch
+---
+
+Small visual adjustment to the the vertical spacing of children when passed to Inline Control Group.
+
+This also cascades down to Radio Group Checkbox Group spacing.
```
Comment (TheSisb) [inline .changeset/two-humans-press.md:8]:
```suggestion
[InlineControlGroup] Adjusted the vertical spacing of `children`.  This change improve the handling of RadioGroup and CheckboxGroup children.
```

File: .changeset/stupid-zebras-wave.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+'@twilio-paste/checkbox': patch
+'@twilio-paste/core': patch
+---
+
+Remove unused value prop from CheckboxGroup as it's now only required in RadioGroup
```
Comment (TheSisb) [inline .changeset/stupid-zebras-wave.md:6]:
```suggestion
[Checkbox] Remove unused `value` prop.
```

File: .changeset/nervous-moles-sleep.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+---
+'@twilio-paste/radio-group': patch
+'@twilio-paste/core': patch
+---
+
+Value is added as a prop used specifically by RadioGroup as it was inheriting its prop interface from InlineContorlGroup which doens't use value at all.
```
Comment (TheSisb) [inline .changeset/nervous-moles-sleep.md:6]: I started reading the changes by only reading the changesets, to see if I could understand the changes. I had a hard time understanding the changesets so I'm providing some suggested improvements.

```suggestion
[RadioGroup] Added optional `value` string prop which this component needs and is no longer on the extended `InlineControlGroup` interface.
```

**Review by TheSisb — COMMENTED** [review]
Really nice changes and improvements. Mostly nits.

File: packages/paste-core/components/form/stories/form.stories.tsx (hunk 1 above)
Comment (SiTaggart) [reply packages/paste-core/components/form/stories/form.stories.tsx:245]: Probably good to give it a try

**Pull request merged**